docs/docs/10.4-test-utils.zh-CN.md

```diff
@@ -0,0 +1,244 @@
+
+> 注意:
+>
+> 你需要让 `window`, `window.document` 和 `window.document.createElement` 全局可用 在你 import React **之前**。
+不然 React 会认为它不能访问 DOM 和类似 `setState` 的方法，然后不工作。
+
+### mockComponent
+
+```javascript
+object mockComponent(
+  function componentClass,
+  [string mockTagName]
+)
+```
+
+传入一个 mocked 组件模块到这个方法来给它增加有用的方法，使它可以被用作 dummy React 组件。代替通常的渲染，组件会成为一个简单的包含了任意被提供的子级的 `<div>` （或者其他 tag 名，如果提供了 `mockTagName`）
+
+### isElement
+
+```javascript
+boolean isElement(
+  ReactElement element
+)
+```
+
+返回 `true` 如果 `element` 是任意的 ReactElement。
+
+### isElementOfType
+
+```javascript
+boolean isElementOfType(
+  ReactElement element,
+  function componentClass
+)
+```
+
+返回 `true` 如果 `element` 是一个类型是 React `componentClass` 的 ReactElement。
+
+### isDOMComponent
+
+```javascript
+boolean isDOMComponent(
+  ReactComponent instance
+)
+```
+
+返回 `true` 如果 `instance` 是一个 DOM 组件 （比如一个 `<div>` 或者 `<span>`）。
+
+### isCompositeComponent
+
+```javascript
+boolean isCompositeComponent(
+  ReactComponent instance
+)
+```
+
+返回 `true` 如果 `instance` 是一个复合组件 （由 `React.createClass()` 创建）。
+
+### isCompositeComponentWithType
+
+```javascript
+boolean isCompositeComponentWithType(
+  ReactComponent instance,
+  function componentClass
+)
+```
+
+返回 `true` 如果 `instance` 是一个类型为 React `componentClass` 的复合组件 （由 `React.createClass()` 创建）。
+
+### findAllInRenderedTree
+
+```javascript
+array findAllInRenderedTree(
+  ReactComponent tree,
+  function test
+)
+```
+
+遍历 `tree` 里所有的组件，并累积所有 `test(component)` 为 `true` 的组件。它本身并没有什么用，但是它被用作其他测试工具的基本元素。
+
+### scryRenderedDOMComponentsWithClass
+
+```javascript
+array scryRenderedDOMComponentsWithClass(
+  ReactComponent tree, string className
+)
+```
+
+在渲染的树中查找所有 DOM组件的类名匹配`className` 的组件实例。
+
+### findRenderedDOMComponentWithClass
+
+```javascript
+ReactComponent findRenderedDOMComponentWithClass(
+  ReactComponent tree,
+  string className
+)
+```
+
+类似 `scryRenderedDOMComponentsWithClass()` 除了只有一个返回结果，并且要么返回这个结果，要么如果还有其他的匹配项就抛出一个异常。
+
+### scryRenderedDOMComponentsWithTag
+
+```javascript
+array scryRenderedDOMComponentsWithTag(
+  ReactComponent tree,
+  string tagName
+)
+```
+
+在渲染的树中查找所有 DOM组件的tag名匹配`tagName` 的组件实例。
+
+### findRenderedDOMComponentWithTag
+
+```javascript
+ReactComponent findRenderedDOMComponentWithTag(
+  ReactComponent tree,
+  string tagName
+)
+```
+
+类似 `scryRenderedDOMComponentsWithTag()` 除了只有一个返回结果，并且要么返回这个结果，要么如果还有其他的匹配项就抛出一个异常。
+
+### scryRenderedComponentsWithType
+
+```javascript
+array scryRenderedComponentsWithType(
+  ReactComponent tree,
+  function componentClass
+)
+```
+
+查找所有类型等于 `componentClass` 的组件实例。
+
+### findRenderedComponentWithType
+
+```javascript
+ReactComponent findRenderedComponentWithType(
+  ReactComponent tree, function componentClass
+)
+```
+
+类似 `scryRenderedComponentsWithType()` 除了只有一个返回结果，并且要么返回这个结果，要么如果还有其他的匹配项就抛出一个异常。
+
+## Shallow rendering（浅渲染）
+
+钱渲染是一个实验性特性，让你渲染一个组件为 "one level deep" 并且断言渲染方法返回的内容，不用担心子组件的行为，它们没有被实例化或者渲染。这个方式不需要一个 DOM。
```

Comment:
@jimfb Maybe there should have a plan to host a top-level Glossary page(en to non en) for non English docs. Cause in Chinese Mainland React community， there are a few kind of different glossary for a same English word， this makes  meanings are quite vague。
As someone said:  `One of the key steps in becoming immersed in an ecosystem is learning its vocabulary`.